### PR TITLE
Adds DataBaseDao + impl.

### PR DESCRIPTION
**Pull request by balag3**
Adds jdbc with sqlite.It work with the product class so far.

Definetely not complete enough to merge into master...

My Plan is to
- adding it for Order class as well
-by implementing the same interfaces as the inMemoDaos creating a Factory
- implement it's service layer and use it with a singleton just like at the inmemory daos.

totally flawed?? :)

### CODE REVIEW

File: src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import com.codecool.webshop.dao.ProductDao;
+import com.codecool.webshop.model.Product;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataBaseProductDaoImpl implements ProductDao {
+
+    static final String DRIVER = "org.sqlite.JDBC";
```
Comment (aron-kovacs) [inline src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java]: These could have private access modifier I suppose

File: src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    static final String URL = "jdbc:sqlite:src/main/java/com/codecool/webshop/database/product.db";
+
+    static Connection conn = null;
```
Comment (aron-kovacs) [inline src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java]: Please don't extract conn, stmt, st and rs fields as instance variables, nor static fields. 
These should have the shortest possible lifecycle, meaning that they will cease to exist when the method finishes it's running.
These objects are handling a heavy logic behind, which means keeping them in memory until the end of the application lifetime is really expensive.

File: src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                "price INTEGER)";
+
+        try {
+            stmt.executeUpdate(sql);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static ResultSet selectAll() {
```
Comment (aron-kovacs) [inline src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java]: Please avoid using static methods as much as possible. Instead, if the DaoImpl class is singleton, you can use simple instance methods.

File: src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    static final String DRIVER = "org.sqlite.JDBC";
+
+    static final String URL = "jdbc:sqlite:src/main/java/com/codecool/webshop/database/product.db";
+
+    static Connection conn = null;
+
+    static Statement stmt = null;
+
+    static PreparedStatement st = null;
```
Comment (aron-kovacs) [inline src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java]: Nevertheless,a great + for using PreparedStatement

File: src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        List<Product> prodList = new ArrayList<>();
+
+        ResultSet rs = selectAll();
+        try {
+            while (rs.next()) {
+                int id = rs.getInt("id");
+                String name = rs.getString("name");
+                int price = rs.getInt("price");
+
+                Product p = new Product();
```
Comment (aron-kovacs) [inline src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java]: You could really use a builder here :-P

**Comment by balag3** [comment]
Did the requested changes. Connection opening and closing are done now from method to method.
Static methods are gone too.

File: src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                if (stmt != null) stmt.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+            try {
+                if (conn != null) conn.close();
```
Comment (aron-kovacs) [inline src/main/java/com/codecool/webshop/dao/impl/DataBaseProductDaoImpl.java]: Connection, PreparedStatement and ResultSet objects implement the same interface (AutoCloseable), have a look at it. So, you could extract a private method for closing and AutoCloseable method, and lines 52-56, 57-61, 83-87, 88-92, 93-97, and so on could replaced by a reference to that method. Let's call it "closeQuietly" :)

**Comment by balag3** [comment]
Yeah I know that you will find it unclean and you were right. :) I was already searching a for a solution for it.

**Comment by balag3** [comment]
Thanks,but I already learned a lot from you! Please keep up this good habit!

**Pull request merged**